llvm/lib/Target/X86/CMakeLists.txt

```diff
@@ -87,6 +87,7 @@ set(sources
   X86VZeroUpper.cpp
   X86WinEHState.cpp
   X86InsertWait.cpp
+  X86VladimirMICounter.cpp
```

Comment:
Please rewrite it as dynamic lib plugin. You may look into #99 for example.

Reply:
Unfortunately, I can’t do this because RegInfo stops working for me. No matter how I tried to collect it in a different way, it did not help.
I suggest leaving everything as is :)

Example Error:
```error
Error opening '/home/sturmannn/llvm-nnsu-2024/build/lib/X86VladimirMICounter.so': /home/sturmannn/llvm-nnsu-2024/build/lib/X86VladimirMICounter.so: undefined symbol: _ZN4llvm3X8612GR64RegClassE
  -load request ignored.
llc: error: run-pass x86-vladimir-mi-counter is not registered.
```

Reply:
> Unfortunately, I can’t do this because RegInfo stops working for me. No matter how I tried to collect it in a different way, it did not help. I suggest leaving everything as is :)
> 
> Example Error:
> 
> ```
> Error opening '/home/sturmannn/llvm-nnsu-2024/build/lib/X86VladimirMICounter.so': /home/sturmannn/llvm-nnsu-2024/build/lib/X86VladimirMICounter.so: undefined symbol: _ZN4llvm3X8612GR64RegClassE
>   -load request ignored.
> llc: error: run-pass x86-vladimir-mi-counter is not registered.
> ```

Idk why u got this, sometimes i got same, but i fixed that. Try debug ur pass at local machine like this
root@overinvest:/home/overinvest/llvm/llvm-nnsu-2024# /home/overinvest/llvm/llvm-nnsu-2024/build/bin/llc -march=x86-64 tested.ll --load=/home/overinvest/llvm/llvm-nnsu-2024/build/lib/X86CountInstBD.so -stop-before=x86-count-machine-instructions -o source_file_test.mir

Reply:
So, as you showed, I also tried to run. But it didn't affect anything... ¯\_(ツ)_/¯

Reply:
Could you please push here your try even if doesn't work properly?
I can look at it at least.

Reply:
[Here is Ubuntu error](https://github.com/NN-complr-tech/llvm-nnsu-2024/actions/runs/8849139632/job/24300401389?pr=105#step:6:475)

Reply:
@m-ly4